File: api/Shop/Model/Category.cs
using System.ComponentModel.DataAnnotations;

namespace Shop.Model;

public class Category
{
    [Key] public uint ID { get; set; }
    [Required] public string Name { get; set; }
    [Required] public int SortOrder { get; set; } = 10;
    public string Description { get; set; }
    public bool Enabled { get; set; } = true;
    public ICollection<Product> Products { get; set; }
}

File: api/shop/model/category.go
package model

import "time"

type Category struct {
	ID          uint   `gorm:"primaryKey" json:"id"`
	Name        string `json:"name"`
	Description string `json:"description"`
	Enabled     *bool  `json:"enabled"`
	CreatedAt   time.Time
	SortOrder   uint `json:"sortOrder"`
}
